app/controllers/cloud_volume_controller.rb

```diff
@@ -19,15 +19,15 @@ def specific_buttons(pressed)
     case pressed
     when 'cloud_volume_attach'
       volume = find_record_with_rbac(CloudVolume, checked_item_id)
-      if !volume.is_available?(:attach_volume) || volume.status != "available"
-        render_flash(_("Cloud Volume \"%{volume_name}\" is not available to be attached to any Instances") % {:volume_name => volume.name}, :error)
+      if !volume.supports?(:attach_volume)
+        render_flash(_("Cloud Volume \"%{volume_name}\" cannot be attached because %{reason}") % {:volume_name => volume.name, :reason => unsupported_reason(:attach_volume)}, :error)
       else
         javascript_redirect(:action => 'attach', :id => checked_item_id)
       end
     when 'cloud_volume_detach'
       volume = find_record_with_rbac(CloudVolume, checked_item_id)
-      if volume.attachments.empty?
```

Comment:
@agrare any idea why the UI is checking the volume's attachments? isn't that the responsibility of either the AvailabilityMixin or the SupportsFeatureMixin? Asking because the tests are now broken and I am trying to fix them in a sane way.